From 6ee878ed799e05b87fe6e4e726d301e1a3d0b754 Mon Sep 17 00:00:00 2001
Message-ID: <6ee878ed799e05b87fe6e4e726d301e1a3d0b754.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 17 Jan 2024 00:08:56 +0000
Subject: [PATCH 25/38] gnu: go-github-com-flynn-noise: Move to (gnu packages
 golang-crypto).

* gnu/packages/golang.scm (go-github-com-flynn-noise): Move from here ...
* gnu/packages/golang-crypto.scm: ... to here.

Change-Id: Ie08a20d57424b0ce5fc10ced2a75b29c8aa3fe7d
---
 gnu/packages/golang-crypto.scm | 26 ++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 27 ---------------------------
 2 files changed, 26 insertions(+), 27 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index c57c525a49..9fec753a35 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -223,6 +223,32 @@ (define-public go-github-com-emersion-go-pgpmail
 messages.")
     (license license:expat)))
 
+(define-public go-github-com-flynn-noise
+  (package
+    (name "go-github-com-flynn-noise")
+    (version "1.0.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/flynn/noise")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1j6phxyqx06wcqxjpin696fkp85s76qcp3i2f7fv6q2fb6618f6y"))))
+    (build-system go-build-system)
+    (arguments
+     '(#:import-path "github.com/flynn/noise"))
+    (propagated-inputs
+     (list go-gopkg-in-check-v1 go-golang-org-x-crypto))
+    (home-page "https://github.com/flynn/noise")
+    (synopsis "Go implementation of the Noise protocol framework")
+    (description "@code{noise} implements the Noise protocol framework.  Noise
+is a low-level framework for building crypto protocols.  Noise protocols
+support mutual and optional authentication, identity hiding, forward secrecy,
+zero round-trip encryption, and other advanced features.")
+    (license license:bsd-3)))
+
 (define-public go-github-com-gaukas-godicttls
   (package
     (name "go-github-com-gaukas-godicttls")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 0ea028295f..87dd184053 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -10390,33 +10390,6 @@ (define-public go-golang-org-x-term
 terminals, as commonly found on Unix systems.")
     (license license:bsd-3)))
 
-(define-public go-github-com-flynn-noise
-  (package
-    (name "go-github-com-flynn-noise")
-    (version "1.0.0")
-    (source
-     (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/flynn/noise")
-             (commit (string-append "v" version))))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32
-         "1j6phxyqx06wcqxjpin696fkp85s76qcp3i2f7fv6q2fb6618f6y"))))
-    (build-system go-build-system)
-    (arguments
-     '(#:import-path "github.com/flynn/noise"))
-    (propagated-inputs
-     (list go-gopkg-in-check-v1 go-golang-org-x-crypto))
-    (home-page "https://github.com/flynn/noise")
-    (synopsis "Go implementation of the Noise protocol framework")
-    (description "@code{noise} implements the Noise protocol framework.  Noise
-is a low-level framework for building crypto protocols.  Noise protocols
-support mutual and optional authentication, identity hiding, forward secrecy,
-zero round-trip encryption, and other advanced features.")
-    (license license:bsd-3)))
-
 (define-public go-github-com-klauspost-compress
   (package
     (name "go-github-com-klauspost-compress")
-- 
2.41.0

